airflow/utils/scarf.py

```diff
@@ -0,0 +1,89 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+import platform
+
+import httpx
+from packaging.version import parse
+
+from airflow import __version__ as airflow_version, settings
+from airflow.configuration import conf
+
+
+def scarf_analytics():
+    if not settings.is_telemetry_collection_enabled():
+        return
+
+    # Exclude pre-releases and dev versions
+    if _version_is_prerelease(airflow_version):
+        return
+
+    scarf_domain = "https://apacheairflow.gateway.scarf.sh/scheduler"
+
+    try:
+        platform_sys, arch = get_platform_info()
+        db_version = get_database_version()
+        db_name = get_database_name()
+        executor = get_executor()
+        python_version = get_python_version()
+
+        scarf_url = (
```

Comment:
probably "better" to use urlencode e.g.
```urlencode({"a": "b", "c": "d and e"})```

Reply:
Should be all strings anyway, but will change it

Reply:
just a suggestion, take it or leave it

Reply:
Took it :) and updated in https://github.com/apache/airflow/pull/39510/commits/982827f725b980586c11facec595ee88da1a4e23